tests/end_to_end_test/main_test.go

```diff
@@ -0,0 +1,81 @@
+package endtoend_test
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/tests/testenv"
+)
+
+var (
+	sharedTestDataDirBase string
+	sharedTestDataDir1    string
+	sharedTestDataDir2    string
+	sharedTestDataDir3    string
+)
+
+func oneTimeSetup() error {
+	var err error
+
+	sharedTestDataDirBase, err = ioutil.TempDir("", "kopia-test")
```

Comment:
Just skimmed through the tests, and AFAICT, these directories are only read from the individual tests and not modified otherwise. And for tests that need specific input setups or so, then `makeScratchDir` below is used.

Is that correct?

Also, `dir1`, `dir2`, `dir3` have approximately similar structures. What's the rationale for generating 3 separate ones? is it an increase in coverage?

Reply:
added more interesting shapes and sizes of the tree along with test API to create more if needed.